api/certificate.go

```diff
@@ -285,158 +304,125 @@ func (c *Client) DeployPFXCertificate(args *DeployPFXArgs) (*DeployPFXResp, erro
-		if lErr != nil {
-			return nil, lErr
-		}
-		return &lCerts[0], nil
-	}
-	return &jsonResp, err
+	mapResp, _ := resp.ToMap()
+	jsonData, _ := json.Marshal(mapResp)
+	var newResp GetCertificateResponse
+	json.Unmarshal(jsonData, &newResp)
+
+	return &newResp, err
 }
 
 func (c *Client) ListCertificates(q map[string]string) ([]GetCertificateResponse, error) {
-	// Set Keyfactor-specific headers
-	headers := &apiHeaders{
-		Headers: []StringTuple{
-			{"x-keyfactor-api-version", "1"},
-			{"x-keyfactor-requested-with", "APIClient"},
-		},
-	}
 
-	// Construct URL query for /Certificates/{ID} requests
-	query := apiQuery{
-		Query: []StringTuple{},
+	type query struct {
+		collectionId         int32
+		pqQueryString        string
+		includeMetadata      bool
+		includeHasPrivateKey bool
+		verbose              int32
+		pqPageReturned       int32
+		pqReturnLimit        int32
+		pqSortField          string
+		pqSortAscending      int32
+		pqIncludeRevoked     bool
+		pqIncludeExpired     bool
+	}
+
+	newQuery := query{
+		collectionId:         0,
+		pqQueryString:        "",
+		includeMetadata:      false,
+		includeHasPrivateKey: false,
+		verbose:              0,
+		pqPageReturned:       0,
+		pqReturnLimit:        0,
+		pqSortField:          "",
+		pqSortAscending:      0,
+		pqIncludeRevoked:     false,
+		pqIncludeExpired:     false,
 	}
-	query.Query = append(query.Query, StringTuple{
-		"includeLocations", "true",
-	})
+
 	searchCollection, ok := q["collection"]
 	if ok {
-		query.Query = append(query.Query, StringTuple{
-			"collectionId", searchCollection,
-		})
+		collectionIdInt, _ := strconv.ParseInt(searchCollection, 10, 64)
+		newQuery.collectionId = int32(collectionIdInt)
 	}
 	subjectName, ok := q["subject"]
 	if ok {
-		query.Query = append(query.Query, StringTuple{
-			"pq.queryString", fmt.Sprintf(`IssuedCN -eq "%s"`, subjectName),
-		})
+		newQuery.pqQueryString = fmt.Sprintf(`IssuedCN -eq "%s"`, subjectName)
 	}
 
-	keyfactorAPIStruct := &request{
-		Method:   "GET",
-		Endpoint: "Certificates",
-		Headers:  headers,
-		Query:    &query,
-		Payload:  nil,
-	}
+	xKeyfactorRequestedWith := "APIClient"
+	xKeyfactorApiVersion := "1"
+
+	configuration := keyfactor_command_client_api.NewConfiguration()
+	apiClient := keyfactor_command_client_api.NewAPIClient(configuration)
+
+	resp, _, err := apiClient.CertificateApi.CertificateQueryCertificates(context.Background()).XKeyfactorRequestedWith(xKeyfactorRequestedWith).CollectionId(newQuery.collectionId).IncludeLocations(true).IncludeMetadata(newQuery.includeMetadata).IncludeHasPrivateKey(newQuery.includeHasPrivateKey).Verbose(newQuery.verbose).XKeyfactorApiVersion(xKeyfactorApiVersion).PqQueryString(newQuery.pqQueryString).PqPageReturned(newQuery.pqPageReturned).PqReturnLimit(newQuery.pqReturnLimit).PqSortField(newQuery.pqSortField).PqSortAscending(newQuery.pqSortAscending).PqIncludeRevoked(newQuery.pqIncludeRevoked).PqIncludeExpired(newQuery.pqIncludeExpired).Execute()
```

Comment:
Can this be expressed a little cleaner?
```
apiClient.CertificateApi.CertificateQueryCertificates(context.Background())\
  .XKeyfactorRequestedWith(xKeyfactorRequestedWith)
```
This is just hard for a human to read.